go/vt/vtgate/evalengine/expressions.go

```diff
@@ -318,40 +365,62 @@ func mergeNumericalTypes(ltype, rtype querypb.Type) querypb.Type {
 	return ltype
 }
 
-func evaluateByType(val *querypb.BindVariable) (EvalResult, error) {
-	switch val.Type {
+func evaluateByTypeSingle(typ querypb.Type, value []byte) (EvalResult, error) {
+	switch typ {
 	case sqltypes.Int64:
-		ival, err := strconv.ParseInt(string(val.Value), 10, 64)
+		ival, err := strconv.ParseInt(string(value), 10, 64)
 		if err != nil {
 			ival = 0
 		}
 		return EvalResult{typ: sqltypes.Int64, numval: uint64(ival)}, nil
 	case sqltypes.Int32:
-		ival, err := strconv.ParseInt(string(val.Value), 10, 32)
+		ival, err := strconv.ParseInt(string(value), 10, 32)
 		if err != nil {
 			ival = 0
 		}
 		return EvalResult{typ: sqltypes.Int32, numval: uint64(ival)}, nil
 	case sqltypes.Uint64:
-		uval, err := strconv.ParseUint(string(val.Value), 10, 64)
+		uval, err := strconv.ParseUint(string(value), 10, 64)
 		if err != nil {
 			uval = 0
 		}
 		return EvalResult{typ: sqltypes.Uint64, numval: uval}, nil
 	case sqltypes.Float64:
-		fval, err := strconv.ParseFloat(string(val.Value), 64)
+		fval, err := strconv.ParseFloat(string(value), 64)
 		if err != nil {
 			fval = 0
 		}
 		return EvalResult{typ: sqltypes.Float64, numval: math.Float64bits(fval)}, nil
 	case sqltypes.VarChar, sqltypes.Text, sqltypes.VarBinary:
-		return EvalResult{typ: sqltypes.VarBinary, bytes: val.Value}, nil
+		return EvalResult{typ: sqltypes.VarBinary, bytes: value}, nil
 	case sqltypes.Time, sqltypes.Datetime, sqltypes.Timestamp, sqltypes.Date:
-		return EvalResult{typ: val.Type, bytes: val.Value}, nil
+		return EvalResult{typ: typ, bytes: value}, nil
 	case sqltypes.Null:
 		return EvalResult{typ: sqltypes.Null}, nil
+	default:
+		return EvalResult{}, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "Type is not supported: %s", typ.String())
+	}
+
+}
+func evaluateByType(val *querypb.BindVariable) (EvalResult, error) {
+	switch val.Type {
+	case querypb.Type_TUPLE:
+		tuple := make([]EvalResult, 0, len(val.Values))
+		for _, value := range val.Values {
+			single, err := evaluateByTypeSingle(value.Type, value.Value)
```

Comment:
this should be `evaluateByType`, so we can support tuples inside of tuples

Reply:
that takes in bindvariable and val.Values range will give query.Value